src/core/client_channel/client_channel.cc

```diff
@@ -1271,35 +1340,26 @@ void ClientChannel::UpdateServiceConfigInDataPlaneLocked(
   ChannelArgs new_args = args.SetObject(this).SetObject(saved_service_config_);
   // Construct filter stack.
   auto new_blackboard = MakeRefCounted<Blackboard>();
-  InterceptionChainBuilder builder(new_args, new_blackboard.get());
-  if (idle_timeout_ != Duration::Zero()) {
-    builder.AddOnServerTrailingMetadata([this](ServerMetadata&) {
-      if (idle_state_.DecreaseCallCount()) StartIdleTimer();
-    });
-  }
-  CoreConfiguration::Get().channel_init().AddToInterceptionChainBuilder(
-      GRPC_CLIENT_CHANNEL, builder);
-  // Add filters returned by the config selector (e.g., xDS HTTP filters).
-  config_selector->AddFilters(builder, blackboard_.get(), new_blackboard.get());
   const bool enable_retries =
       !channel_args_.WantMinimalStack() &&
       channel_args_.GetBool(GRPC_ARG_ENABLE_RETRIES).value_or(true);
   if (enable_retries) {
     RetryInterceptor::UpdateBlackboard(*saved_service_config_,
                                        blackboard_.get(), new_blackboard.get());
-    builder.Add<RetryInterceptor>(nullptr);
   }
-  blackboard_ = std::move(new_blackboard);
-  // Create call destination.
-  auto top_of_stack_call_destination = builder.Build(call_destination_);
-  // Send result to data plane.
-  if (!top_of_stack_call_destination.ok()) {
-    resolver_data_for_calls_.Set(MaybeRewriteIllegalStatusCode(
-        top_of_stack_call_destination.status(), "channel construction"));
-  } else {
-    resolver_data_for_calls_.Set(ResolverDataForCalls{
-        std::move(config_selector), std::move(*top_of_stack_call_destination)});
+  std::function<void(ServerMetadata&)> on_server_trailing_metadata;
```

Comment:
AnyInvocable

Reply:
I would have used `AnyInvocable`, but it needs to be copyable.  The builder may be used to build more than one filter chain, and this callback needs to be registered with each one.